src/liquid/components/ld-cookie-consent/ld-cookie-consent.tsx

```diff
@@ -0,0 +1,789 @@
+ * @part preferences-buttons - preferences buttons container
+ * @part preferences-category-description - description of category specific category
+ * @part preferences-category-table - cookie table
+ * @part preferences-category-table-container - cookie table container
+ * @part preferences-category-table-td - cookie table data cell
+ * @part preferences-category-table-th - cookie table header cell
+ * @part preferences-category-table-tr - cookie table row
+ * @part preferences-category-title - title of category specific category
+ * @part preferences-category-toggle - toggle of category specific category
+ * @part preferences-description - preferences description container
+ * @part preferences-header - preferences header containing logo and title
+ * @part preferences-logo-container - logo container in the preferences header
+ * @part preferences-privacy-policy-notice - privacy notice in modal layer footer
+ * @part preferences-privacy-policy-notice-container - container for privacy notice slot
+ * @part preferences-title - the preferences title displayed in the modal layer header
+ */
+@Component({
+  tag: 'ld-cookie-consent',
+  styleUrl: 'ld-cookie-consent.shadow.css',
+  shadow: true,
+})
+export class LdCookieConsent {
+  @Element() el: HTMLElement
+
+  /** Component settings object to be merged with the default options (optionally stringified). */
+  @Prop() settings!: Partial<LdCookieConsentConfig> | string
+
+  /** Emitted when the cookie consent disclaimer is shown. */
+  @Event() ldCookieConsentDisclaimerShow: EventEmitter<undefined>
+
+  /** Emitted after loading saved consent from local storage. */
+  @Event() ldCookieSavedConsentLoad: EventEmitter<{
+    acceptedCategories: Set<string>
+    consentDate: Date
+    rejectedCategories: Set<string>
+    revision: number
+  }>
+
+  /** Emitted when the cookie consent preferences modal layer is shown. */
+  @Event() ldCookieConsentPreferencesShow: EventEmitter<undefined>
+
+  /** Emitted after clearing cookies which happens after saving setting. */
+  @Event() ldCookieConsentAutoclearCookies: EventEmitter<undefined>
+
+  /** Emitted after saving cookies preferences to local storage with accepted categories. */
+  @Event() ldCookieConsentSave: EventEmitter<{
+    acceptedCategories: Set<string>
+    consentDate: Date
+    rejectedCategories: Set<string>
+    revision: number
+  }>
+
+  /** Emitted after activating scripts (only emitted if at least one script has been activated). */
+  @Event() ldCookieConsentActivateScripts: EventEmitter<undefined>
+
+  @State() private isDisclaimerVisible = false
+  @State() private allCategories: Set<string> = new Set()
+  @State() private preselectedCategories: Set<string> = new Set()
+  @State() private savedConsent: {
+    acceptedCategories: Set<string>
+    consentDate: Date
+    rejectedCategories: Set<string>
+    revision: number
+  } | null
+  @State() private selectedCategories: Set<string> = new Set()
```

Comment:
Using `Set`s results in quite some boilerplate code. I'm not sure if it's worth it just to describe that these contain only unique values.

Reply:
Quite the contrary. Keyed collections have been around quite some time now (since ES6 I believe) and we should use them, because they reduce boilerplate and have better performance. 
Here's a link to the ES6 specs: https://262.ecma-international.org/8.0/#sec-set-objects 
Our target in the tsconfig is set to `"es2017"`.